### PR TITLE
Added progress bar in uploading raw cosv files

### LINKED ISSUE #2678: Support progress bar in uploading raw cosv files

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- added endpoint to upload a single raw cosv file (just for a case)
- supported progress bar for raw cosv files via `application/x-ndjson`
- supported content length for raw cosv file dto
- updated storage for tests source snapshot to use a new property for temp files

It closes #2678 

*Note:* there is an issue with displaying the progress bar for ZIP files. Will be fixed in scope of #2685

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/TestsSourceSnapshotStorage.kt [hunk 1]
```diff
@@ -25,39 +24,47 @@ import kotlin.io.path.*
...
+
     /**
      * @param request
      * @return [TestFilesContent] filled with test files
      */
-    fun getTestContent(request: TestFilesRequest): Mono<TestFilesContent> {
-        val tmpSourceDir = createTempDirectory("source-")
-        val tmpArchive = createTempFile(tmpSourceDir, "archive-", ARCHIVE_EXTENSION)
-        val sourceContent = download(request.testsSourceSnapshot)
-            .map { DefaultDataBufferFactory.sharedInstance.wrap(it) }
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/TestsSourceSnapshotStorage.kt:40]: dont quite understand where did `DefaultDataBufferFactory` and `DataBufferUtils` disappear to

**Review by kgevorkyan — APPROVED** [review]

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/TestsSourceSnapshotStorage.kt (hunk 1 above)
Comment (nulls) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/TestsSourceSnapshotStorage.kt:40]: re-use existed method `collectToFile` which uses `DefaultDataBufferFactory` inside

**Comment by nulls** [comment]
Very long processing one-by-one.
upload-cosv-one-by-one
upload-cosv-one-by-one-2

**Comment by nulls** [comment]
upload-cosv-one-by-one-3

but invalid works for ZIP:
upload-cosv-one-by-one-4


probably needs to move extracting to a separate task...

**Pull request merged**